server/clcachesrv.py

```diff
@@ -46,6 +49,16 @@ def _onPathChange(self, handle, filename, events, error):
             logging.debug("invalidating cached hashsum for %s", os.path.join(handle.path, filename))
             del watchedDirectory[filename]
 
+    def isExcluded(self, dirname):
+        # as long as we do not hate more than _MAXCACHE regex we can
+        # rely on the internal cacheing of re.match
+        for pattern in vars(self._args)['exclude']:
+            if re.search(pattern, dirname):
+                logging.debug("NOT watching %s", dirname)
+                return True
+
+        return False
```

Comment:
You could shorten this to

```
excluded = any(re.search(pattern, dirname) for pattern in vars(self._args)['exclude']))
if excluded:
    logging.debug("NOT watching %s", dirname)
return excluded
```

...but two more important things also come to my mind:

1. Do you really mean to use `re.search` here? It will match the pattern *anywhere* in the dir, e.g. the pattern `\\build\\` would match `C:\build\mydata` as well as `C:\foo\bar\build\yoyodyne` etc.. Maybe `re.match` would be more like what one would expect?

2.  Windows paths are case insensitive, so I imagine that the regular expression should match in a case insensitive way, too -- it appears you can achieve this by passing `re.IGNORECASE` as the third argument to `re.search` (or `re.match`).

Having written this and noticing that backslashes are both very common for Windows paths *and* need escaping, I wonder whether regular expressions are maybe overkill. Maybe it would be sufficient to allow specifying prefixes to match against and then we just call `os.path.normcase` to normalise the paths and see whether one starts with another?